src/utils/styles.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//TODO do we need to export this to clients (index.js) ?
+
+// like mui sx(slot)/class merging but simpler with less features
+// TODO use their system ? But it's named unstable_composeClasses so not supported?
+export const makeComposeClasses =
+    (generateGlobalClass) => (classes, ruleName) =>
+        [generateGlobalClass(ruleName), classes[ruleName]]
+            .filter((x) => x)
+            .join(' ');
+
+export const toNestedGlobalSelectors = (styles, generateGlobalClass) =>
+    Object.fromEntries(
+        Object.entries(styles).map(([k, v]) => [
+            `& .${generateGlobalClass(k)}`,
```

Comment:
Why do we have to use this ?

Reply:
I mean, we can't get rid of classNames ?

Reply:
Yes, but almost everytime, we stylise the component itself and not nested components.
Or I missed something ?